framework/include/meshlet_helpers.hpp

```diff
@@ -0,0 +1,174 @@
+	 *	@param	aMaxVertices		The maximum number of vertices of a meshlet.
+	 *	@param	aMaxIndices			The maximum number of indices of a meshlet.
+	 *	@param	aCombineSubmeshes	If submeshes should be combined into a single vertex/index buffer.
+	 */
+	std::vector<meshlet> divide_into_meshlets(const std::vector<std::tuple<avk::resource_ownership<gvk::model_t>, std::vector<mesh_index_t>>>& aModels,
+		const uint32_t aMaxVertices = 64, const uint32_t aMaxIndices = 378, const bool aCombineSubmeshes = true);
+
+	/** Divides the given models into meshlets using the given callback function.
+	 *  @param	aModelsAndMeshletIndices				All the models and associated meshes that should be divided into meshlets.
+	 *  @param	aMeshletDivision	Callback used to divide meshes into meshlets with a maximum number of vertices and indices.
+	 *								It can either receive the vertices and indices or just the indices depending on your specific needs.
+	 *								Additionally it provides the model and an optional mesh index if more data is needed. If no mesh index is provided then the meshes were combined beforehand.
+	 *								Must not grab ownership of the model, the model will be assigned to the meshlets after its execution.\n\n
+	 *								The callback has a specific layout, optional parameters can be omitted, but all of them need to be provided in the following order.
+	 *								Parameters in definition order:
+	 *								 - const std::vector<glm::vec3>& tVertices:		optional	The vertices of the mesh or combined meshes of the model
+	 *								 - const std::vector<uint32_t>& tIndices:  		mandatory	The indices of the mesh or combined meshes of the model
+	 *								 - const model_t& tModel:						mandatory	The model these meshlets are generated from
+	 *								 - std::optional<mesh_index_t> tMeshIndex: 		mandatory	The optional mesh index. If no value is provided, it means the meshes of the model are combined into a single vertex and index buffer
+	 *								 - uint32_t tMaxVertices:						mandatory	The maximum number of vertices that are allowed in a single meshlet.
+	 *								 - uint32_t tMaxIndices:						mandatory	The maximum number of indices that are allowed in a single meshlet.
+	 *								Return value:
+	 *								 - std::vector<meshlet>										The generated meshlets.\n\n
+	 *								Example:
+	 *								@code
+	 *								[](const std::vector<glm::vec3>& tVertices, const std::vector<uint32_t>& aIndices,
+	 *											const model_t& aModel, std::optional<mesh_index_t> aMeshIndex,
+	 *											uint32_t aMaxVertices, uint32_t aMaxIndices) {
+	 *									std::vector<gvk::meshlet> generatedMeshlets;
+	 *									// Do your meshlet division here
+	 *									return generatedMeshlets;
+	 *								}
+	 *								@endcode
+	 *	@param	aMaxVertices		The maximum number of vertices of a meshlet.
+	 *	@param	aMaxIndices			The maximum number of indices of a meshlet.
+	 *	@param	aCombineSubmeshes	If submeshes should be combined into a single vertex/index buffer.
```

Comment:
I think, it would be nicer if this parameter was positioned before `aMaxVertices` and `aMaxIndices` because it is more related to the models and meshes. If it is the last parameter, one could assume that it is more related to vertices/indices limits.

Please adapt in all places (code and comments).